templates/server/config.ru.erb

```diff
@@ -79,9 +79,8 @@ class Puppet3Compat
       env['QUERY_STRING'] = Rack::Utils.build_query(@req.params)
     end
 
-    if @api =~ /^file_(content|bucket_file)/ && @req.get?
-      env["HTTP_ACCEPT"] = "binary"
-    elsif @api == "file_bucket_file" && (@req.post? || @req.put?)
+    env["HTTP_ACCEPT"] = env["HTTP_ACCEPT"].split(/\s*,\s*/).map { |a| a.sub(/^raw|s$/, 'binary') }.join(', ')
```

Comment:
In the code you linked there's also a distinct which I think should be here as well before joining again.